ast/src/parsed/display.rs

```diff
@@ -919,6 +919,19 @@ fn format_list_of_types<E: Display>(types: &[Type<E>]) -> String {
         .to_string()
 }
 
+/// Formats a list of types to be used as values for type arguments
+/// and puts them in angle brackets.
+/// Puts the last item in parentheses if it ends in `>` to avoid parser problems.
+pub fn format_type_args<E: Display>(args: &[Type<E>]) -> String {
+    let mut items = args.iter().map(|arg| format!("{arg}")).rev().peekable();
```

Comment:
It's actually quite interesting that this works: The iterator is reversed, then the peek_mut evaluates the first (i.e. last) element in the list, modifies it and then the iterator is reversed again...